pkg/cloudprovider/providers/azure/azure_instances.go

```diff
@@ -119,6 +121,13 @@ func (az *Cloud) NodeAddresses(ctx context.Context, name types.NodeName) ([]v1.N
 				})
 			}
 		}
+
+		if len(addresses) == 1 {
```

Comment:
a little confused about the number, why it's 1 not 0, more comments here?

Reply:
Hostname is already included above